### ISSUE TITLE
Streamline package updates for `amphtml`

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#33959]
**Background:**

Today, `amphtml` uses https://github.com/apps/forking-renovate to regularly update all `devDependencies` (and some `dependencies`) listed in the root `package.json` file and various sub-package level files. As of today, there have been close to [~1500 update PRs](https://github.com/ampproject/amphtml/pulls/renovate-bot), all of which have been merged manually. Other than a very small number of breaking changes, all these PRs were rubber-stamp approved and merged after all CI checks were green.

**Goal:**

The purpose of this issue is to streamline the upgrade process, eliminate unnecessary manual toil, and require human intervention only while approving critical runtime dependencies or addressing breaking changes. Here's a proposal:

- Enable automatic merging for renovate PRs via the `automerge` config setting (see [docs](https://docs.renovatebot.com/automerge-configuration/))
- Since AMP's PRs need one approving review, use the [`renovate-approve`](https://github.com/apps/renovate-approve) app to auto-approve PRs that pass CI checks
- Set `renovate-approve` as an owner of `package.json` and `package-lock.json` files
- Do not enable automatic merging of upgrades to AMP's `dependencies` (shipped with the runtime)

With this, most incoming upgrade PRs that pass all CI checks will be automatically approved and merged.

**Comment by rsimha** [comment]
@kristoferbaxter @jridgewell @dvoytenko Since y'all are top-level owners, please comment if you have any concerns, and thumbs-up if the plan SGTY.

/cc @ampproject/wg-infra

**Comment by jridgewell** [comment]
As long as they're devDeps, seems fine.

**Comment by kristoferbaxter** [comment]
This seems fine to me.

**Comment by rsimha** [comment]
> As long as they're devDeps, seems fine.

Yes, this will be enabled only for renovate [groups](https://github.com/ampproject/amphtml/blob/e5c57a647118dfff7f6a2e81bcf5a624550504f1/.renovaterc.json#L23-L97) that cover `devDepdendencies`, so there's no risk of default-merging regular `dependencies` or having this auto-applied to new groups added in future.
